packages/core/src/util/txInspector.ts

```diff
@@ -270,57 +267,75 @@ export const withdrawalInspector: WithdrawalInspector = (tx) =>
   tx.body.withdrawals?.length ? BigIntMath.sum(tx.body.withdrawals.map(({ quantity }) => quantity)) : 0n;
 
 /**
- * Inspects a transaction for minted or burned assets.
+ * Matching criteria functor definition. Ths functor encodes a selection criteria for minted/burned assets.
+ * For example to get all minted assets the following criteria could be provided:
+ *  (quantity: bigint) => quantity > 0.
+ */
+export interface MatchQuantityCriteria {
+  (quantity: bigint): boolean;
+}
+
+/**
+ * Inspects a transaction for minted/burned assets that match a given quantity criteria.
  *
- * @param {TxAlonzo} tx transaction to inspect.
- * @returns {Value} A collection of minted and burned assets grouped by policy Id.
+ * @param matchQuantityCriteria A functor that represents a selection criteria for minted/burned assets. Will
+ * return <tt>true</tt> if the given criteria was meet; otherwise; <tt>false</tt>.
+ * @returns A collection with the assets that match the given criteria.
  */
-export const assetsMintedInspector: AssetsMintedInspector = (tx) => {
-  const assets: AssetsMintedInspection = { burned: [], minted: [] };
-  const scriptMap = new Map();
-
-  if (!tx.body.mint) return assets;
-
-  // Scripts can be embedded in transaction auxiliary data and/or the transaction witness set. If this transaction
-  // was built by this client the script will be present in the witness set, however, if this transaction was
-  // queried from a remote repository that doesn't fetch the witness data of the transaction we can still check
-  // if the script is present in the auxiliary data.
-  const scripts = [...(tx.auxiliaryData?.body?.scripts || []), ...(tx.witness?.scripts || [])];
-
-  for (const script of scripts) {
-    let policyId;
-    switch (script.__type) {
-      case ScriptType.Native:
-        policyId = nativeScriptPolicyId(script);
-        if (scriptMap.has(policyId)) continue;
-        scriptMap.set(policyId, script);
-        break;
-      case ScriptType.Plutus: // TODO: Add support for plutus minting scripts.
-      default:
-      // scripts of unknown type will be ignored.
+export const mintInspector =
+  (matchQuantityCriteria: MatchQuantityCriteria): AssetsMintedInspector =>
+  (tx) => {
+    const assets: AssetsMintedInspection = [];
+    const scriptMap = new Map();
+
+    if (!tx.body.mint) return assets;
```

Comment:
I did not find this corner case in the doc block. Please add it if you think it brings value. Thanks!